torch/csrc/inductor/aot_inductor_tensor.h

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+#include <torch/csrc/inductor/aot_inductor_utils.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum {
+  kAotInductorCPU = 0,
+  kAotInductorCUDA = 1,
+} AotInductorDeviceType;
+
+typedef enum {
+  kAotInductorByte = 0,
+  kAotInductorChar = 1,
+  kAotInductorShort = 2,
+  kAotInductorInt = 3,
+  kAotInductorLong = 4,
+  kAotInductorHalf = 5,
+  kAotInductorFloat = 6,
+  kAotInductorDouble = 7,
+  kAotInductorComplexHalf = 8,
+  kAotInductorComplexFloat = 9,
+  kAotInductorComplexDouble = 10,
+  kAotInductorBool = 11,
+  kAotInductorQInt8 = 12,
+  kAotInductorQUInt8 = 13,
+  kAotInductorQInt32 = 14,
+  kAotInductorBFloat16 = 15,
+} AotInductorScalarType;
+
+typedef struct {
+  AotInductorDeviceType device_type;
+  int8_t device_id;
+} AotInductorDevice;
+
+// Bare minumum tensor struct to interchange data between AOTInducotor
+// generated shared library and libtorch to avoid ABI compatibility issues
+typedef struct {
+  void* data_ptr;
+  void* deleter;
+  AotInductorDevice device;
+  AotInductorScalarType dtype;
+  int64_t ndim;
+  const int64_t* sizes;
+  const int64_t* strides;
+  int64_t storage_offset;
+} AotInductorTensor;
```

Comment:
If these structs are part of the public ABI, I am a bit nervous about them. You have no padding on the structs, so it's not possible to add new fields to these structs without it being an ABI breaking change. A common alternative is to provide an opaque function for allocating and filling the struct, which you can then successively upgrade the functions (`_v2`, etc) if you need to add new fields later.